### PR TITLE
3.2: Packs libraries alongside executable in function.zip

### PR DESCRIPTION
**Pull request by Karm**
Clean backport of https://github.com/quarkusio/quarkus/pull/35718

### CODE REVIEW

**Review by gsmet — APPROVED** [review]

**Comment by gsmet** [comment]
@rsvoboda this is a 3.2 backport so I let you have a look before merging.

File: extensions/amazon-lambda/common-deployment/src/main/java/io/quarkus/amazon/lambda/deployment/FunctionZipProcessor.java [hunk 1]
```diff
@@ -137,8 +142,26 @@ public void nativeZip(OutputTargetBuildItem target,
                 }
             }
             addZipEntry(zip, nativeImage.getPath(), executableName, 0755);
+
+            final GraalVM.Version graalVMVersion = nativeImageRunner.getBuildRunner().getGraalVMVersion();
+            if (graalVMVersion.compareTo(GraalVM.Version.VERSION_23_0_0) >= 0) {
+                // See https://github.com/oracle/graal/issues/4921
+                try (DirectoryStream<Path> sharedLibs = Files.newDirectoryStream(nativeImage.getPath().getParent(),
+                        "*.{so,dll}")) {
```
Comment (rsvoboda) [inline extensions/amazon-lambda/common-deployment/src/main/java/io/quarkus/amazon/lambda/deployment/FunctionZipProcessor.java:150]: I know the use-case if for Amazon Lambda, but should this also include `jnilib` to cover macOS?
CC @Karm

Comment (Karm) [reply extensions/amazon-lambda/common-deployment/src/main/java/io/quarkus/amazon/lambda/deployment/FunctionZipProcessor.java:150]: I have no access to any mac device, so I don't know. @galderz

Comment (galderz) [reply extensions/amazon-lambda/common-deployment/src/main/java/io/quarkus/amazon/lambda/deployment/FunctionZipProcessor.java:150]: AFAIK `dylib` is the extension on macos.

Comment (rsvoboda) [reply extensions/amazon-lambda/common-deployment/src/main/java/io/quarkus/amazon/lambda/deployment/FunctionZipProcessor.java:150]: ah, system based libraries, not the ones in JAR files.

Anyway, should be `dylib` added?

Comment (galderz) [reply extensions/amazon-lambda/common-deployment/src/main/java/io/quarkus/amazon/lambda/deployment/FunctionZipProcessor.java:150]: > Anyway, should be dylib added?

I guess if this functions can run on macOS then yeah, but don't know how feasible that is.

**Pull request merged**